packages/squiggle-lang/src/fr/plot.ts

```diff
@@ -107,13 +110,28 @@ const _assertYScaleNotDateScale = (yScale: Scale | null) => {
   }
 };
 
+function formatXPoints(
```

Comment:
We might well want to do more work on this in the components, but it seems good to have this data be as good as possible here.

This ensures that the xPoints are uniq, ordered, and within range.